torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp

```diff
@@ -274,262 +656,387 @@ void scheduleReduction(
 
     // Do multiple reductions per block
     if (rparams.multiple_reds_per_blk) {
-      // Reduction Splits
-      //      [outputs, |rF-Leftover, X-Warp, rf-Unroll|]
-      // Idx:     0     |   1(-1)      2(-2)     3(-1) |
-      //                --------------------------------
-      //                Reduction Dimensions
-      red_tv->split(reduce_axis, rparams.loop_unroll);
-      red_tv->split(
-          reduce_axis, NamedScalar::getParallelDim(ParallelType::TIDx));
-
-      // Output Splits
-      //      [|Out-Leftover, Out-PerBlock|, <Reduction Dims>]
-      // Idx:  |     0             1      |   2(-2) -- 3(-1)
-      //       ----------------------------
-      //       Output Dimensions
-      if (has_iter_axis) {
+      if (rparams.reduction_unroll) {
+        // Fastest dim, multiple reductions per block
+        // Output Dimensions
+        // [x-BIDx, x-TIDy
+        //  0       1
+        //
+        //  Reduction Dimensions
+        //  rF-Remain, rf-Unswitch, rf-Unroll, X-TIDx]
+        //  2 (-4)     3 (-3)       4 (-2)     5 (-1)
+
         red_tv->split(
-            iter_axis, NamedScalar::getParallelDim(ParallelType::TIDy));
-        for (auto iter_tv : outs_of_red) {
-          iter_tv->split(
+            reduce_axis, NamedScalar::getParallelDim(ParallelType::TIDx));
+        red_tv->split(reduce_axis, rparams.loop_unroll);
+        red_tv->split(reduce_axis, 1);
```

Comment:
Why are splitting by 1? I am assuming for `rf-Unswitch`. Maybe add comment for clarity.

Reply:
done